### PR TITLE
i#3479: Fix RUN_IN_RELEASE set of tests for AArch64.

### LINKED ISSUE #3479: Run at least a few release-build tests on Travis
In light of a serious regression crashing all drcachesim/drmemtrace uses in #3478 which was not detected pre-commit but in our post-commit CDash tests, this issue proposes adding at least some subset of tests to be run in release build on Travis.  The reason we don't run all of them is to keep the pre-commit time down to something reasonable.

I'm proposing giving a few key tests "TRAVIS" labels and run with "-L TRAVIS".  We're already doing the release build on Travis so a few tests won't add too much time.

For reference on how long it takes to build vs run tests, here are the numbers on my CDash Linux machine:

debug-internal-64
build: 1:54
tests: 10:28

release-external-64
build: 1:42
tests: 6:13

### PR DESCRIPTION
**Pull request by hgreving2304**
Make common.flags test independent of architecture in RUN_IN_RELEASE set of tests.

Fixes #3479

### CODE REVIEW

File: suite/tests/CMakeLists.txt [hunk 1]
```diff
@@ -3751,7 +3751,7 @@ endforeach(run)
 # A subset of release tests has been added and labeled with RUN_IN_RELEASE. It focuses on
 # Google's use of the drcachesim client. We do not want to extend the Travis regression
 # time by too much and keep the following list concise.
-if (NOT DEBUG AND UNIX AND X64 AND NOT CMAKE_COMPILER_IS_CLANG)
+if (NOT DEBUG AND UNIX AND X64 AND X86 AND NOT CMAKE_COMPILER_IS_CLANG)
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: > Excludes AArch64 from running release-external-64 tests on Travis.

This does not sound like the right solution.  A RUN_IN_RELEASE tag should be cross-platform. 
 Just use the name `common.${control_flags}`.

Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: Ok, that's better. Thanks.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -3754,7 +3754,7 @@ endforeach(run)
 if (NOT DEBUG AND UNIX AND X64 AND NOT CMAKE_COMPILER_IS_CLANG)
   set_tests_properties(
     samples_proj
-    code_api|common.eflags
+    code_api|common.${control_flags}
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt]: LGTM but best to first test locally in a cross-compile release build for A64 (are all of the tests below on A64?  If not best to split the list (or (longer-term?) move to setting these on individual tests so we don't have this separate control point to keep in sync).

Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: I don't have any A64 machine. @AssadHashmi , could you try out this patch?

Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: @hgreving2304 no need to run, just build: you should be able to use the same cross-compiler packages used by Travis.

Comment (hgreving2304) [reply suite/tests/CMakeLists.txt]: True. Ok, done, works.

**Comment by AssadHashmi** [comment]
I'm getting the same cmake error with this patch.
This is the command on AArch64:
cmake -DBUILD_TESTS=ON ..

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: Or, just temporarily enable BUILD_TESTS for the arm-release-external-64 build on Travis.

**Comment by derekbruening** [comment]
> I'm getting the same cmake error with this patch.

It must be a different error, right?  Could you paste the error?

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (derekbruening) [reply suite/tests/CMakeLists.txt]: (Though for the future it is useful to have a local cross-compile build dir so I would suggest setting it up regardless.)

**Comment by hgreving2304** [comment]
(also looking at it, I ran arm32 instead of a64...)

**Comment by AssadHashmi** [comment]
Current head
```
$ git log --abbrev-commit  --pretty='format:%h %an, %ar, %s'| head
93785cea Derek Bruening, 2 hours ago, Change -cache_commit_increment to uint_size to accept prefixes (#3506)
39aaaf88 Hendrik Greving, 21 hours ago, i#1312 AVX-512 support: Partially change REG_ into DR_REG_ in encode. (#3503)
f2cf6815 Derek Bruening, 22 hours ago, i#3314: Free drmodtrack user memory on parsing errors (#3500)
```

Patch
```
$ git diff
diff --git a/suite/tests/CMakeLists.txt b/suite/tests/CMakeLists.txt
index 8a7fe73d..85ad71e8 100644
--- a/suite/tests/CMakeLists.txt
+++ b/suite/tests/CMakeLists.txt
@@ -3754,7 +3754,7 @@ endforeach(run)
 if (NOT DEBUG AND UNIX AND X64 AND NOT CMAKE_COMPILER_IS_CLANG)
   set_tests_properties(
     samples_proj
-    code_api|common.eflags
+    code_api|common.${control_flags}
     code_api|tool.drcachesim.simple
     code_api|tool.drcachesim.simple-config-file
     code_api|tool.drcachesim.TLB-simple
```

Error
```
$ tail build/my_cmake.log
-- WARNING: vera++ not found: disabling code style checks
-- Processing tests and generating expected output patterns
CMake Error at suite/tests/CMakeLists.txt:3755 (set_tests_properties):
  set_tests_properties Can not find test to add properties to:
  code_api|tool.drcachesim.threads


-- Configuring incomplete, errors occurred!
```

**Comment by hgreving2304** [comment]
> Current head
> 
> ```
> $ git log --abbrev-commit  --pretty='format:%h %an, %ar, %s'| head
> 93785cea Derek Bruening, 2 hours ago, Change -cache_commit_increment to uint_size to accept prefixes (#3506)
> 39aaaf88 Hendrik Greving, 21 hours ago, i#1312 AVX-512 support: Partially change REG_ into DR_REG_ in encode. (#3503)
> f2cf6815 Derek Bruening, 22 hours ago, i#3314: Free drmodtrack user memory on parsing errors (#3500)
> ```
> Patch
> 
> ```
> $ git diff
> diff --git a/suite/tests/CMakeLists.txt b/suite/tests/CMakeLists.txt
> index 8a7fe73d..85ad71e8 100644
> --- a/suite/tests/CMakeLists.txt
> +++ b/suite/tests/CMakeLists.txt
> @@ -3754,7 +3754,7 @@ endforeach(run)
>  if (NOT DEBUG AND UNIX AND X64 AND NOT CMAKE_COMPILER_IS_CLANG)
>    set_tests_properties(
>      samples_proj
> -    code_api|common.eflags
> +    code_api|common.${control_flags}
>      code_api|tool.drcachesim.simple
>      code_api|tool.drcachesim.simple-config-file
>      code_api|tool.drcachesim.TLB-simple
> ```
> Error
> 
> ```
> $ tail build/my_cmake.log
> -- WARNING: vera++ not found: disabling code style checks
> -- Processing tests and generating expected output patterns
> CMake Error at suite/tests/CMakeLists.txt:3755 (set_tests_properties):
>   set_tests_properties Can not find test to add properties to:
>   code_api|tool.drcachesim.threads
> 
> 
> -- Configuring incomplete, errors occurred!
> ```

Try now.

**Comment by derekbruening** [comment]
I would suggest adding a comment to change this list to be a property added to tests individually which would eliminate this type of problem altogether.

**Comment by AssadHashmi** [comment]
> > Error
> > ```
> > $ tail build/my_cmake.log
> > -- WARNING: vera++ not found: disabling code style checks
> > -- Processing tests and generating expected output patterns
> > CMake Error at suite/tests/CMakeLists.txt:3755 (set_tests_properties):
> >   set_tests_properties Can not find test to add properties to:
> >   code_api|tool.drcachesim.threads
> > 
> > 
> > -- Configuring incomplete, errors occurred!
> > ```
> 
> Try now.

That works, thanks.

**Pull request merged**